File: frontend/src/api/submissions.ts
import api from './client';

export interface SubmitTaskPayload {
  city_id: string;
  name: string;
  description?: string;
  address?: string;
  task_description?: string;
  verification_hint?: string;
  verification_type: string;
  category: string;
  vibe_tags: string[];
  price_level?: number | null;
  avg_duration_minutes?: number | null;
  pro_tips?: string;
}

export interface Submission {
  id: string;
  submitted_by: string;
  submitter_username: string;
  city_id: string;
  city_name: string;
  name: string;
  description: string | null;
  address: string | null;
  task_description: string | null;
  verification_hint: string | null;
  verification_type: string;
  category: string;
  vibe_tags: string[];
  price_level: number | null;
  avg_duration_minutes: number | null;
  pro_tips: string | null;
  status: string;
  rejection_reason: string | null;
  created_at: string;
}

export async function submitTask(payload: SubmitTaskPayload) {
  const { data } = await api.post<Submission>('/submissions/', payload);
  return data;
}

export async function getMySubmissions() {
  const { data } = await api.get<Submission[]>('/submissions/mine');
  return data;
}

export async function getSubmissions(status = 'pending') {
  const { data } = await api.get<Submission[]>('/submissions/', { params: { status } });
  return data;
}

export async function reviewSubmission(id: string, action: 'approve' | 'reject', opts?: { rejection_reason?: string; xp?: number }) {
  const { data } = await api.patch<Submission>(`/submissions/${id}`, { action, ...opts });
  return data;
}

File: app/schemas/submissions.py
"""Pydantic schemas for task submissions."""
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class SubmitTaskRequest(BaseModel):
    city_id: UUID
    name: str = Field(..., min_length=2, max_length=255)
    description: str | None = None
    address: str | None = None
    task_description: str | None = None
    verification_hint: str | None = None
    verification_type: str = "photo"
    category: str = Field(..., min_length=1, max_length=50)
    vibe_tags: list[str] = []
    price_level: int | None = Field(None, ge=1, le=4)
    avg_duration_minutes: int | None = Field(None, ge=1)
    pro_tips: str | None = None


class SubmissionResponse(BaseModel):
    id: UUID
    submitted_by: UUID
    submitter_username: str
    city_id: UUID
    city_name: str
    name: str
    description: str | None
    address: str | None
    task_description: str | None
    verification_hint: str | None
    verification_type: str
    category: str
    vibe_tags: list[str]
    price_level: int | None
    avg_duration_minutes: int | None
    pro_tips: str | None
    status: str
    rejection_reason: str | None
    created_at: datetime

    class Config:
        from_attributes = True


class ReviewSubmissionRequest(BaseModel):
    action: str = Field(..., pattern="^(approve|reject)$")
    rejection_reason: str | None = None
    xp: int | None = Field(None, ge=0)
